dsl/camel-jbang/camel-jbang-plugin-k/src/main/java/org/apache/camel/dsl/jbang/core/commands/k/support/SourceHelper.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.dsl.jbang.core.commands.k.support;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.camel.main.KameletMain;
+
+public final class SourceHelper {
+
+    private SourceHelper() {
+        //prevent instantiation of utility class.
+    }
+
+    public static Set<String> dependencies(List<Source> sources) throws Exception {
+        List<String> files = new ArrayList<>();
+
+        for (Source s : sources) {
+            if (s.local() && !s.path().startsWith("classpath:")) {
+                // get the absolute path for a local file
+                files.add("file://" + new File(s.path()).getAbsolutePath());
+            } else {
+                files.add(s.path());
+            }
+        }
+
+        KameletMain main = new KameletMain();
+        DependenciesDownloadListener downloadListener = new DependenciesDownloadListener(sources);
```

Comment:
just a note, there is a [PR to revert](https://github.com/apache/camel/pull/13404) this.

Reply:
thx, this is pretty much a leftover that I was supposed to remove, Initially I did try to use the download listeners but realized that in fact we do no need to download anything to perform route inspection.